### PR TITLE
fix(components): [el-affix] fix target resize bug

### PR DESCRIPTION
**Pull request by midcu**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/affix/src/affix.vue [hunk 1]
```diff
@@ -131,6 +131,7 @@ export default defineComponent({
 
     useEventListener(scrollContainer, 'scroll', onScroll)
     useResizeObserver(root, () => update())
+    useResizeObserver(target, () => update())
```
Comment (JeremyWuuuuu) [inline packages/components/affix/src/affix.vue:134]: I'm just being curious. Why can't we just use `useResizeObserver(target, update)` here?

Comment (sxzz) [reply packages/components/affix/src/affix.vue:134]: To make it clear that this is a callback

**Review by sxzz — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
@midcu I left a comment, would you mind address that so we can get this merged? Even though the code is pretty simple but we still need you to change the code as requested.

**Comment by sxzz** [comment]
@JeremyWuuuuu Sorry, but i don't think so.

**Comment by CFMJ** [comment]
问题并没有被解决  2.2.0版本还是有一样的问题

**Pull request merged**